### ISSUE TITLE
[IMPROVEMENT] Make environment check script recognize iscsid.socket enable instead of iscsid.server only  

### ISSUE THREAD

**Issue by innobead** [longhorn/longhorn#5380]
## Is your improvement request related to a feature? Please describe (👍 if you like this request)

Currently, in the environment_check script, it will check if iscsid.server is running or not by systemd; however, if iscsid.service is not enabled by default and not running, it doesn't mean iscsid is not working as expected. Probably it's due to it is not enabled by default, so it will be only activated by `iscsid.socket` which is enabled by default.

## Describe the solution you'd like

Have another optional check to see if `iscsid.socket` is enabled. If it's, the environment should be ready for iSCSI client operations.

**Comment by innobead** [comment]
cc @longhorn/qa

**Comment by jfmatth** [comment]
Agreed.  I installed Centos8-stream and just installing the iscsi-initiator-utils will not make Longhorn work.  LH will install, but won't provision volumes and fails strangely.  

Not sure how to solve on the check, but i had to enable the service and socket to make sure it was working.

oh yea, and make sure FirewallD is disabled too.

**Comment by innobead** [comment]
@derekbit is helping this while working on SPDK env check.

**Comment by roger-ryao** [comment]
**Verified on master-head 20230629**

- longhorn master-head https://github.com/longhorn/longhorn/commit/6b56bb2b7240aaa814b853c10e40e705cfd713a8
- longhorn-manager master-head https://github.com/longhorn/longhorn-manager/commit/9bc7e074d3b6105ab95234140ddb89b0dad9cbdd

**Result Passed**

```bash
ryao@ryao-laptop:~/Downloads/ssh_key> curl -sSfL https://raw.githubusercontent.com/longhorn/longhorn/master/scripts/environment_check.sh | bash
[INFO]  Required dependencies 'kubectl jq mktemp' are installed.
[INFO]  All nodes have unique hostnames.
[INFO]  Waiting for longhorn-environment-check pods to become ready (0/1)...
[INFO]  All longhorn-environment-check pods are ready (1/1).
[INFO]  MountPropagation is enabled
[INFO]  Checking iscsid...
[ERROR] Neither iscsid.service nor iscsid.socket is not running on ryao-master-w1-cef4a091-qtwwt
[INFO]  Checking multipathd...
[INFO]  Checking packages...
[ERROR] nfs-common is not found in ryao-master-w1-cef4a091-qtwwt.
[INFO]  Cleaning up longhorn-environment-check pods...
[INFO]  Cleanup completed.
```
